src/lambda_streams/Animals.java

```diff
@@ -1,61 +1,222 @@
 package lambda_streams;
 
 import java.util.*;
+import java.util.stream.Collectors;
 
 class Animals {
     //This class will be using streams.
     static List<String> animals = Arrays.asList("peacoCK","rabbit","chiwawa","OranguTAN","vipeR","cobra","paNDa","bUffalo","DeeR","maLLard");
 
     public static void main(String[] args) {
+
+        lambdaListBooleanProcessor lambdaUpperFirst = (animalList, mutate) -> {
+            if(mutate){
+                return animals = animalList.stream().map(animal -> animal.substring(0, 1).toUpperCase() + animal.substring(1).toLowerCase()).collect(Collectors.toList());
+            } else {
+                List<String> updatedAnimals = animalList;
+                return updatedAnimals.stream().map(animal -> animal.substring(0, 1).toUpperCase() + animal.substring(1).toLowerCase()).collect(Collectors.toList());
```

Comment:
Lines 14 and 17 contain the same code (though using different Collections. Consider if this were code for a system, and someone later needed to change how the list is compiled. If they aren't careful, they may update one of these instances, but not the other, introducing a defect. How might this code be refactored such that the only code within the If/Else is the code that is different. This would ensure there is one, and only one line of code responsible for adjusting the Caps of the animal names.